### ISSUE TITLE
Make Yato a library type target in cmake

### ISSUE THREAD

**Issue by ckorikov** [agruzdev/Yato#2]
Why not to use

```cmake
add_library(Yato INTERFACE)
```

instead of  `add_custom_target` ?

https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries

**Comment by agruzdev** [comment]
As I see the INTERFACE target doesn't compile the provided source files (since 3.19 it can contain them), but the Yato library can be either header-only or contain .cpp files depending on build flags. So I would like to build it as a separate target, when it has sources.

**Comment by ckorikov** [comment]
I see. Actually, I asked because it could be pretty convenient to use the library in the following way.

```cmake
add_library(MyLib)
target_link_libraries(MyLib PRIVATE Yato)
```

According to cmake specification it seems to be impossible for custom targets.

https://cmake.org/cmake/help/latest/command/target_link_libraries.html

**Comment by agruzdev** [comment]
If you use only the core headers, then you dont need to add link to the Yato at all, since it is header-only library, just `include_directories( )` would be enough.
If you use the config or actors module, the link the YatoConfig or YatoActors libraries directly, like this:
```
add_library(MyLib)
target_link_libraries(MyLib PRIVATE YatoConfig)
```

Let me know if you think it is still inconvenient

**Comment by ckorikov** [comment]
Right, but `target_link_libraries` also enables usage requirements propagation from a child to a parent target. It can be only headers in the header-only library or something else in other cases. From the user side, it looks like a general way to connect a dependency that hides its compilation details.

**Comment by agruzdev** [comment]
I got your point now. I'll think how I can improve this...
